src/main/java/io/cdap/plugin/successfactors/connector/SuccessFactorsConnectorConfig.java

```diff
@@ -0,0 +1,131 @@
+  private static final Logger LOG = LoggerFactory.getLogger(SuccessFactorsConnectorConfig.class);
+
+  @Name(UNAME)
+  @Macro
+  @Description("SAP SuccessFactors Username for user authentication.")
+  private final String username;
+
+  @Name(PASSWORD)
+  @Macro
+  @Description("SAP SuccessFactors password for user authentication.")
+  private final String password;
+
+  @Macro
+  @Name(BASE_URL)
+  @Description("SuccessFactors Base URL.")
+  private final String baseURL;
+
+  public SuccessFactorsConnectorConfig(String username, String password, String baseURL) {
+    this.username = username;
+    this.password = password;
+    this.baseURL = baseURL;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  public String getBaseURL() {
+    return baseURL;
+  }
+
+  public void validateBasicCredentials(FailureCollector failureCollector) {
+
+    if (SuccessFactorsUtil.isNullOrEmpty(getUsername()) && !containsMacro(UNAME)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(SAP_SUCCESSFACTORS_USERNAME);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(UNAME);
+    }
+    if (SuccessFactorsUtil.isNullOrEmpty(getPassword()) && !containsMacro(PASSWORD)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(SAP_SUCCESSFACTORS_PASSWORD);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(PASSWORD);
+    }
+    if (SuccessFactorsUtil.isNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey(SAP_SUCCESSFACTORS_BASE_URL);
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(BASE_URL);
+    }
+    if (SuccessFactorsUtil.isNotNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      if (HttpUrl.parse(getBaseURL()) == null) {
+        String errMsg = ResourceConstants.ERR_INVALID_BASE_URL.getMsgForKey(SAP_SUCCESSFACTORS_BASE_URL);
+        failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(BASE_URL);
+      }
+    }
+  }
+
+  /**
+   * Method to validate the credential fields.
+   */
+  public void validateConnection(FailureCollector collector) {
+    SuccessFactorsTransporter successFactorsHttpClient = new SuccessFactorsTransporter(getUsername(), getPassword());
+    URL testerURL = HttpUrl.parse(getBaseURL()).newBuilder().build().url();
+    SuccessFactorsResponseContainer responseContainer = null;
+    try {
+      responseContainer =
+        successFactorsHttpClient.callSuccessFactorsEntity(testerURL, MediaType.APPLICATION_JSON, TEST);
+    } catch (TransportException e) {
+      LOG.error("Unable to fetch the response", e);
+      collector.addFailure("Unable to call SuccessFatorsEntity", "Please check the values");
+      throw collector.getOrThrowException();
```

Comment:
Similar to the other error cases in this method, I don't think you need throw collector.getOrThrowException() here.

Reply:
removed